### PR TITLE
Migrating the existing configuration and removing the legacy configuration modules

### PR DESCRIPTION
**Pull request by ssheng**
## Description

Migrating the existing configuration and removing the legacy configuration modules.

### Removed CLI command `bentoml config` ###

The command `config` is removed from the `bentoml` CLI. Use the environment variable `BENTOML_CONFIG` to specify the config file instead.

### Deprecated `bentoml yatai-service-start --repo-base-url ###

The CLI option `--repo-base-url` is deprecated and replaced with more specific options `--repository-type`. The values for `--repository-type` can be `FILE_SYSTEM`, `S3`, or `GCS`. Based on the selection, more specific options can be used to customized the more specific behaviors.

- `--file-system-directory`: Specifies the directory path for the file system repository type
- `--s3-url`: Specifies the S3 URL for the S3 repository type
- `--s3-endpoint-url`: S3 Endpoint URL is used for deploying with storage services that are compatible with Amazon S3, such as MinIO
- `--gcs-url`: Specifies the GCS URL for the GCS repository type

### Removed support for `bentoml.cfg` based configuration ###

Configuration parameters specified with `bentoml.cfg` will be replaced by the `bentoml.yml` based configurations. In addition, new sections `bento_bundle`, `yatai`, and `logging` are added to the `bentoml.yml` template and schema. Legacy `bentoml.cfg` is no longer merged with `bentoml.yml` for backward compatibility support. Documentation will be updated in a follow-up pull request.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/cli/yatai_service.py [hunk 1]
```diff
@@ -61,32 +80,93 @@ def add_yatai_service_sub_command(cli):
...
+        s3_url,
+        s3_endpoint_url,
+        gcs_url,
     ):
+        from bentoml.utils.s3 import is_s3_url
+        from bentoml.utils.gcs import is_gcs_url
+
+        if repo_base_url:
+            logger.warning(
+                "Option --repo-base-url has been deprecated but is still supported "
```
Comment (parano) [inline bentoml/cli/yatai_service.py:145]: 👍

File: bentoml/configuration/__init__.py [hunk 2]
```diff
@@ -15,21 +15,16 @@
 import os
 import logging
 from functools import lru_cache
-from pathlib import Path
 
 from bentoml import __version__, _version as version_mod
-from bentoml.exceptions import BentoMLConfigException
-from bentoml.configuration.configparser import BentoMLConfigParser
+
+# from bentoml.configuration.containers import BentoMLContainer
```
Comment (ssheng) [inline bentoml/configuration/__init__.py:21]: Will remove

Comment (ssheng) [inline bentoml/configuration/__init__.py:21]:
```suggestion
```

File: bentoml/saved_bundle/config.py [hunk 3]
```diff
@@ -74,14 +75,22 @@ def _get_artifacts_list(bento_service):
 
 
 class SavedBundleConfig(object):
-    def __init__(self, bento_service=None, kind="BentoService"):
+    @inject
+    def __init__(
+        self,
+        bento_service=None,
+        kind="BentoService",
+        bentoml_version: str = Provide[
```
Comment (ssheng) [inline bentoml/saved_bundle/config.py:83]:
```suggestion
        bentoml_deployment_version: str = Provide[
```

File: bentoml/service/env.py [hunk 4]
```diff
@@ -162,7 +163,13 @@ def __init__(
         conda_env_yml_file: str = None,
         setup_sh: str = None,
         docker_base_image: str = None,
+        default_docker_base_image: str = Provide[
+            BentoMLContainer.config.bento_bundle.default_docker_base_image
+        ],
         zipimport_archives: List[str] = None,
+        bentoml_version: str = Provide[
```
Comment (ssheng) [inline bentoml/service/env.py:170]:
```suggestion
        bentoml_deployment_version: str = Provide[
```

**Review by yubozhao — APPROVED** [review]

**Pull request merged**